cypress-io/cypress
# feat: IATR non-failure run statuses

- Closes #24848

### User facing changelog


n/a, merging to a feature branch

### Additional details


For some run statuses on the Debug tab, we need to display specific information, sometimes in addition to displaying the failing tests for debugging. This PR adds UI for the following run states:

- `CANCELLED`
- `PASSED`
- `ERRORED`
- `NOTESTS`
- `TIMEDOUT`
- `OVERLIMIT`

### Steps to test


Ideally, you should create a run on your local Cloud instance for each of the statuses listed above. I was only able to actually re-create a few of them end to end (I couldn't re-create `TIMEDOUT` or `OVERLIMIT`). I would suggest checking out the tests that I added [here](https://github.com/cypress-io/cypress/compare/feature/IATR-M0...24848-non-failure-run-statuses?expand=1#diff-f97e5572c31990dca87154da4f715e65acc797e63c864d5996134b33fbd5694c) and viewing the Percy snapshots that were generated from those tests and comparing those to the [Figma](https://www.figma.com/file/PAOozuCxtfetqP8fUVvg0y/branch/QkCp2F69X0qyLo7qnYQ67e/Cypress-App?node-id=21180%3A190683&t=mgpzpKNCFo6r0Y3a-1).

There's an issue with the tests for `PASSED` and `OVERLIMIT` where in the test they appear towards the top of the viewport instead of further down where they should be. However, if you actually view these states on the Debug tab, they show up correctly. There should be a `flex-grow` class on these that create the correct height for the elements but it's not working in the tests.

### How has the user experience changed?


See Percy diff

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
